### PR TITLE
[CORE] Add metadata exclusion list for native write

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

This pr adds a new config `spark.gluten.sql.native.writeFilesColumnMetadataWhiteList` to allow native write files even if there are some metadata. By default, set with `comment` which is quite common if the source table column has comment and the write files column reference it directly. 

## How was this patch tested?

add test

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @PHILO-HE thank you

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 1]
```diff
@@ -1326,6 +1326,14 @@ object GlutenConfig {
       .booleanConf
       .createOptional
 
+  val NATIVE_WRITE_FILES_COLUMN_METADATA_WHITE_LIST =
+    buildConf("spark.gluten.sql.native.writeFilesColumnMetadataWhiteList")
```
Comment (PHILO-HE) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:1330]: "whiteList" looks misleading, as the listed items are actually removed from output. Maybe, `xxx.WriteColumnMetadataExclusionList`?

Comment (ulysses-you) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala]: I'm fine with this new name, changed it

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**